src/connection.js

```diff
@@ -747,7 +779,11 @@ class Connection extends EventEmitter {
   saveTransaction(callback, name) {
     const transaction = new Transaction(name);
     if (this.config.options.tdsVersion < '7_2') {
-      return this.execSqlBatch(new Request('SAVE TRAN ' + transaction.name, callback));
+      const self = this;
+      return this.execSqlBatch(new Request('SAVE TRAN ' + transaction.name, function() {
```

Comment:
In all these cases where you used `const self = this;` to capture the `this` context outside the callback, you can also use arrow functions now that you're no longer passing along the `this` context of the callback:

```js
return this.execSqlBatch(new Request('SAVE TRAN ' + transaction.name, () => {
  this.transactionDepth++;
  return callback.apply(null, arguments);
}));
```